### ISSUE TITLE
tag list already have some build bug

### ISSUE THREAD

**Issue by youxiachai** [hexojs/hexo#1019]
in hexo 3.0 rc1 
I have 50 posts, just 10 posts build right.

pre post format like this
```
title: 从kindle3到kindle voyage
date: 2014-11-23 17:32:53
tags: xx
---
```

build  exception show like this

![qq 20150130101928](https://cloud.githubusercontent.com/assets/929502/5970158/8e85ca9c-a869-11e4-8a42-a2c2f8fe950d.jpg)

pre 3.0@beta.3 tag list not right ,but post create not this problem.

**hexo 2.8.3 not this problem**

**Comment by tommy351** [comment]
It seems the front-matter is not parsed correctly. I'll take a look at this problem.

**Comment by iissnan** [comment]
I have encountered the same problem on Windows 8 with Hexo 3.0.0-rc.1

```
hexo: 3.0.0-rc.1
os: Windows_NT 6.1.7601 win32 x64
http_parser: 1.0
node: 0.10.29
v8: 3.14.5.9
ares: 1.9.0-DEV
uv: 0.10.27
zlib: 1.2.3
modules: 11
openssl: 1.0.1h
```

Meanwhile, this problem does not reproduce on Ubuntu 12.04 LTS. But on Ubuntu 12.04, condition helpers like `is_home()/is_post()` work incorrectly. All of their values evaluated to false.

**Comment by tommy351** [comment]
I can't reproduce this problem on Mac too. I'll try on Windows later.

**Comment by tommy351** [comment]
It seems this bug only appear on Windows. I'll fix it later.

**Comment by tommy351** [comment]
The reason of this problem is Windows uses `\r\n` as line ending and I forgot to escape them. This bug has been fixed in commit 33bf64a
